iib/web/models.py

```diff
@@ -190,6 +204,79 @@ def add_state(self, state, state_reason):
         db.session.flush()
         self.request_state_id = request_state.id
 
+    def to_json(self, verbose=True):
+        states = [
```

Comment:
If `verbose` is not set, use the `state` relationship to get the latest state, otherwise, continue with your approach.

See https://github.com/release-engineering/cachito/pull/87 which fixes this.